### PR TITLE
feat: manager applications in organization scope

### PR DESCRIPTION
**Pull request by leo220yuyaodog**
### Organization admin
1.Organization admin can manager own applications. 

image

2.And they can't modify the organization of the application

image

###  Global admin
1.Same as before. The global admin can manager all application.

image

2.Can modify the organization of applications.

image

## Providers
There is a kind of providers, their owner is "admin". They can be shared for all applications. 
### Organization admin
Manager their own provider

image

### Global admin
1. Manager all providers. 
2. Can modify the organzaition of providers. 
3. What'more, setting the "admin" providesr.

image

## What providers can be used by an application?
1. Providers belong to admin.

image

2. Provider belong to the organization of the application.

image

image

close: #1262

### CODE REVIEW

File: object/application_item.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func (application *Application) GetSignupItemRule(itemName string) string {
 }
 
 func (application *Application) getAllPromptedProviderItems() []*ProviderItem {
-	res := []*ProviderItem{}
+	var res []*ProviderItem
```
Comment (hsluoyz) [inline object/application_item.go:90]: Will this work on zero length?

File: object/token.go [hunk 2]
```diff
@@ -678,7 +678,7 @@ func GetWechatMiniProgramToken(application *Application, code string, host strin
 			ErrorDescription: "the application does not support wechat mini program",
 		}
 	}
-	provider := GetProvider(util.GetId(mpProvider.Name))
+	provider := GetProviderByName(mpProvider.Name)
```
Comment (hsluoyz) [inline object/token.go:681]: Why change this?

File: object/provider_item.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ package object
 
 type ProviderItem struct {
 	Name      string    `json:"name"`
+	Owner     string    `json:"owner"`
```
Comment (hsluoyz) [inline object/provider_item.go:19]: Before name

File: authz/authz.go [hunk 4]
```diff
@@ -142,7 +142,7 @@ func IsAllowed(subOwner string, subName string, method string, urlPath string, o
 
 	userId := fmt.Sprintf("%s/%s", subOwner, subName)
 	user := object.GetUser(userId)
-	if user != nil && user.IsAdmin && (subOwner == objOwner || (objOwner == "admin" && subOwner == objName)) {
+	if user != nil && user.IsAdmin && (subOwner == objOwner || (objOwner == "admin")) {
```
Comment (hsluoyz) [inline authz/authz.go:145]: Don't allow now

File: web/src/App.js [hunk 5]
```diff
@@ -565,10 +564,11 @@ class App extends Component {
...
           <Route exact path="/adapters/:organizationName/:adapterName" render={(props) => this.renderLoginIfNotLoggedIn(<AdapterEditPage account={this.state.account} {...props} />)} />
           <Route exact path="/providers" render={(props) => this.renderLoginIfNotLoggedIn(<ProviderListPage account={this.state.account} {...props} />)} />
-          <Route exact path="/providers/:providerName" render={(props) => this.renderLoginIfNotLoggedIn(<ProviderEditPage account={this.state.account} {...props} />)} />
-          <Route exact path="/providers/:organizationName/:providerName" render={(props) => this.renderLoginIfNotLoggedIn(<ProviderEditPage account={this.state.account} {...props} />)} />
+          {/* <Route exact path="/providers/:providerName" render={(props) => this.renderLoginIfNotLoggedIn(<ProviderEditPage account={this.state.account} {...props} />)} />*/}
+          <Route exact path="/providers/:owner/:providerName" render={(props) => this.renderLoginIfNotLoggedIn(<ProviderEditPage account={this.state.account} {...props} />)} />
```
Comment (hsluoyz) [inline web/src/App.js:568]: owner --> organizationName

File: web/src/backend/ProviderBackend.js [hunk 6]
```diff
@@ -24,6 +24,10 @@ export function getProviders(owner, page = "", pageSize = "", field = "", value
   }).then(res => res.json());
 }
 
+export function getAdminAndOrganizationProviders(organization) {
```
Comment (hsluoyz) [inline web/src/backend/ProviderBackend.js:27]: We don't need `getAdminAndOrganizationProviders`

File: web/src/ApplicationListPage.js [hunk 7]
```diff
@@ -53,15 +69,15 @@ class ApplicationListPage extends BaseListPage {
       redirectUris: ["http://localhost:9000/callback"],
       tokenFormat: "JWT",
       expireInHours: 24 * 7,
-      formOffset: 8,
+      formOffset: 2,
```
Comment (hsluoyz) [inline web/src/ApplicationListPage.js:72]: Why change CSS?

File: web/src/ApplicationListPage.js [hunk 8]
```diff
@@ -23,11 +23,27 @@ import i18next from "i18next";
...
+      data: [],
+      pagination: {
+        current: 1,
+        pageSize: 10,
+      },
+      loading: false,
+      searchText: "",
+      searchedColumn: "",
+    };
+  }
```
Comment (hsluoyz) [inline web/src/ApplicationListPage.js:40]: new line

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by leo220yuyaodog** [comment]
@hsluoyz  plz review

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**